internal/collector/collector.go

```diff
@@ -177,6 +176,7 @@ func (f *Factory) NewCollectorContainer(inputs internalobs.Inputs, outputs inter
 	}
 	collector.Env = []v1.EnvVar{
 		{Name: "COLLECTOR_CONF_HASH", Value: f.ConfigHash},
+		{Name: "SECRETS_HASH", Value: f.Secrets.HashMD5()},
```

Comment:
Is this hash output somewhere? If not, then it might be an idea to store it in a Pod annotation instead (or do both and populate the environment variable from the Pod annotation).